Source/WebCore/rendering/RenderBox.cpp

```diff
@@ -1466,7 +1481,10 @@ bool RenderBox::hasTrimmedMargin(PhysicalDirection physicalDirection) const
 #endif
     ASSERT(!needsLayout());
 
-    if (physicalDirection == PhysicalDirection::Top)
+    auto canUsePhysicalToFlowRelativeMapping = [&]() {
+        return isFlexItem();
+    };
+    if ((physicalDirection == PhysicalDirection::Top || physicalDirection == PhysicalDirection::Right) && canUsePhysicalToFlowRelativeMapping())
```

Comment:
so this is gonna assert anyway if called with unsupported content?

Reply:
Yeah that's not quite right, thanks for catching. After taking another look I think this can be removed since any asserts should be guarded by the logic of `rendererContainingBlockhasMarginTrim` (or whatever the name of that function will become) before calling `hasTrimmedMargin`